src/net/tcp/stream.rs

```diff
@@ -148,6 +97,78 @@ impl TcpStream {
         }))
     }
 
+    /// Creates a new TCP stream connected to the specified address.
+    ///
+    /// This method will create a new TCP socket and attempt to connect it to the `addr`
+    /// provided. The returned future will be resolved once the stream has successfully
+    /// connected, or it will return an error if one occurs.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # fn main() -> std::io::Result<()> { async_std::task::block_on(async {
+    /// #
+    /// use async_std::net::TcpStream;
+    ///
+    /// let addr = "127.0.0.1".parse().unwrap();
+    /// let stream = TcpStream::connect_to(addr).await?;
+    /// #
+    /// # Ok(()) }) }
+    /// ```
+    pub async fn connect_to(addr: SocketAddr) -> io::Result<TcpStream> {
```

Comment:
I'd prefer not making this method public since the standard library doesn't have its equivalent. That does not mean I'm against having additional methods per se, but rather that it'd be easier to punt those decisions to post-1.0 release (Sep 26, 2019).

Reply:
Sure

Reply:
Fixed